docs/decisions/0002-initial-api-specification.rst

```diff
@@ -0,0 +1,436 @@
+should not be enabled via the API (though perhaps are possible via Django Admin).
+
+Subsidy Transactions
+********************
+**/api/v1/subsidies/[subsidy-uuid]/transactions/**
+The root URL for reading aggregate and transaction-level data about the transactions
+associated with a subsidy.
+
+GET (list) enterprise subsidy transactions
+==========================================
+**/api/v1/subsidies/[subsidy-uuid]/transactions/?include_aggregates={true,false}**
+
+This endpoint can answer questions like:
+
+- Tell me all of the transactions for a given subsidy.
+- Within a subsidy, give me all transactions for a specific user or content key.
+- Given a user id and content key, tell me if a successful/committed transaction exists, that is,
+  has the user previously redeemed this subsidy for this content?
+
+Inputs
+------
+
+- ``subsidy-uuid`` (URL path, required): The uuid (primary key) of the subsidy for which transactions should be listed.
+- ``include_aggregates`` (query param, optional): Specifies if aggregates (quantities, number of transactions) should be
+  returned as part of the paginated response.  Defaults to ``true``.
+- ``user_id`` (query param, optional): If present, filters returned transactions and/or aggregates to only
+  those associated with the specified ``user_id`` value.
+- ``content_key`` (query param, optional): If present, filters returned transactions and/or aggregates to only
+  those associated with the specified ``content_key`` value.
+
+Outputs
+-------
+Returns a paginated list of aggregate and transaction data:
+
+::
+
+   {
+       'previous': null,
+       'next': '...',
+       'count': 'blah',
+       'aggregates': {
+           'total_quantity': 12350,
+           'unit': 'USD_CENTS',
+           'remaining_balance': 987650
+       },
+       'results': [
+           {
+               'uuid': 'the-transaction-uuid',
+               'status': 'completed',  TODO: enumerate valid statuses
+               'idempotency_key': 'the-idempotency-key',
+               'user_id': 54321,
+               'content_key': 'demox_1234+2T2023',
+               'quantity': 19900,
+               'unit': 'USD_CENTS',
+               'reference_id': 1234,
+               'reference_table': 'enrollments',
+               'subsidy_access_policy_uuid': 'a-policy-uuid',
+               'metadata': {...},
+               'created': 'created-datetime',
+               'modified': 'modified-datetime',
+               'reversal': null
+           }
+       ]
+   }
+
+Permissions
+-----------
+
+enterprise_learner
+  The transaction data returned should be filtered to only those related to the requesting user, within
+  the subsidy specified in the request's URL path.
+
+enterprise_admin
+  Should return the requested subsidy transactions only if the requesting user has implicit (JWT) or explicit (DB-defined)
+  ``enterprise_admin`` role assigned for the requested subsidy's enterprise.
```

Comment:
Curious in general about these non-operator roles across all the subsidy/ledger endpoints: will we ever expect that a customer or learner will interact directly with these endpoints?  My understanding was that access would either be granted to ECS/engineering (openedx_operator), or the backend service enterprise-access (also openedx_operator?).

Reply:
I've tried to structure it so that non-operators/non-staff/non-services can only read relevant data.  Will add a note in `Consequences`.